src/node_crypto.cc

```diff
@@ -699,6 +699,32 @@ void SecureContext::AddRootCerts(const FunctionCallbackInfo<Value>& args) {
 
   assert(sc->ca_store_ == NULL);
 
+  if (args.Length() == 2) {
+    char* caFile = NULL;
+    char* caPath = NULL;
+    if (args[0]->IsString()) {
+      String::Utf8Value file(args[0]);
+      caFile = *file;
```

Comment:
This is unsafe; you're assigning a pointer to memory that goes out of scope immediately afterwards.  Happens a few lines below as well.